### DISCUSSION TITLE
Turbo command to start Vite and APIs at the same time

### DISCUSSION THREAD

**Discussion by leo-petrucci** [vercel/turborepo#7851] [Help] [answered]
### Summary

The way I understand Vercel expects me to use Turborepo is to name different scripts in different workspaces `start` or `build` and then running `turbo start` or `turbo build` to run them at once.

In the setup I'm currently working on we use Turbo to run dev tasks with a single command. One example is starting a front-end as well as two APIs it depends on.

I do this with this config:

```
{
  "$schema": "https://turbo.build/schema.json",
  "globalDependencies": ["**/.env.*local"],
  "pipeline": {
    "frontend": {
      "dependsOn": [
        "frontend#frontend:dev",
        "^product-api#products:dev",
        "^auth-api#auth:dev"
      ]
      "dependsOn": []
    },
    "frontend#frontend:dev": {},
    "product-api#products:dev": {},
    "auth-api#auth:dev": {},
  }
}
```

We have around 5 different micro-frontends that are all independent from each other and have their own `turbo` configs.

This used to work great up until a while ago, but something changed as mentioned in #7427 . Now our Vite frontend immediately stops after starting.

Unfortunately when I add `"persistent": true` to `"frontend#frontend:dev": {},` I start getting the following errors:
```
Error:   × "frontend#frontend:dev" is a persistent task, "^product-api#products:dev" cannot depend on it
   ╭─[turbo.json:6:1]
 6 │       "dependsOn": [
 7 │         "frontend#frontend:dev",
   ·         ─────────────┬─────────────
   ·                      ╰── persistent task
 8 │         "^product-api#products:dev",
   ╰────
```

Is there any way around this?

**Comment by anthonyshew** [comment, marked as answer]
Looking at the issue, it appears the problem with the Vite dev server has been fixed. Can you confirm with the latest version of `turbo`?

As far as depending on persistent tasks goes, Turborepo is working correctly here. We can't depend on a persistent task since it will never exit.

In the bigger picture, it's seemingly impossible for `turbo` to know when to start the frontends according to the state of your backends (since they don't exit).

Said another way, the big question is: How would Turborepo know when your backends are prepared so that it can start the frontends?

What I recommend you do today is use two separate `turbo` processes, in two terminals. First, run a `turbo run dev:backends`. Second, once those are ready, run a `turbo run dev:frontends` command.

**Reply by leo-petrucci** [reply]
You're right, latest version did fix it!

> As far as depending on persistent tasks goes, Turborepo is working correctly here. We can't depend on a persistent task since it will never exit.

Yeah sorry, I meant more in the context of the Vite command. Didn't realise it stopping was a separate issue.

> Said another way, the big question is: How would Turborepo know when your backends are prepared so that it can start the frontends?

Strangely enough this does seem to work fine with Serverless Framework, but it's kind of a non-issue since it's just used for development.
